analyzer/src/main/kotlin/managers/Pub.kt

```diff
@@ -0,0 +1,533 @@
+            val type = packageInfo["source"].textValueOrEmpty();
+            val description = packageInfo["description"];
+            val packageName = description["name"].textValueOrEmpty();
+            val url = description["url"].textValueOrEmpty();
+            val resolvedRef = packageInfo["resolved-ref"].textValueOrEmpty();
+
+            if(type == "hosted" && url.isNotEmpty()){
+                // packages with source set to "hosted" and "url" key in description set to "https://pub.dartlang.org",
+                // path should be resolved to "hosted/pub.dartlang.org/packageName-packageVersion"
+                path = "hosted/" + url.replace("https://","") + "/" + packageName + "-" + packageVersion;
+            } else if(type == "git" && resolvedRef.isNotEmpty()){
+                // packages with source set to "git" and a "resolved-ref" key in description set to a gitHash,
+                // should be resolved to "git/packageName-gitHash"
+                path = "git/" + packageName + "-" + resolvedRef;
+            } else {
+                // not supported type
+                path = "/";
+
+                log.error { "Could not find projectRoot of '$packageName'" }
+            }
+
+            val artifactRootDir = File(
+                pubCacheRoot,
+                path
+            )
+
+            return artifactRootDir;
+        }
+    }
+
+    private val reader = PubCacheReader();
+
+    override fun resolveDependencies(definitionFile: File): ProjectAnalyzerResult? {
+        val workingDir = definitionFile.parentFile
+
+        val manifest = yamlMapper.readTree(definitionFile)
+        val hasDependencies = manifest.fields().asSequence().any { (key, value) ->
+            key.startsWith("dependencies") && value.count() > 0
+        }
+
+        val (packages, scopes) = if (hasDependencies) {
+            installDependencies(workingDir)
+
+            log.info { "Reading $PUB_LOCK_FILE file in $workingDir " }
+            val lockFile = yamlMapper.readTree(File(workingDir, PUB_LOCK_FILE))
+
+            log.info { "Reading lockfile success. " }
+
+            var packages = parseInstalledPackages(lockFile)
+
+            log.info { "parseInstalledPackages success. " }
+
+            val scopes = sortedSetOf(
+                parseScope("dependencies", manifest, lockFile, packages),
+                parseScope("dev_dependencies", manifest, lockFile, packages)
+            )
+
+            Pair(packages, scopes)
+        } else {
+            Pair(emptyMap(), sortedSetOf())
+        }
+
+        log.info { "Reading ${definitionFile.name} file in $workingDir " }
+
+        val project = parseProject(definitionFile, scopes)
+
+        return ProjectAnalyzerResult(project, packages.values.map { it.toCuratedPackage() }.toSortedSet())
+
+    }
+
+    private fun parseScope(
+        scopeName: String, manifest: JsonNode, lockFile: JsonNode, packages: Map<Identifier, Package>
+    ): Scope {
```

Comment:
If the function signature does not fit in one line, put each argument in a separate line, following the Kotlin style guide.